### DISCUSSION TITLE
Resume terminal state when codespace is stopped

### DISCUSSION THREAD

**Discussion by brooke-hamilton** [community/community#4773] [Codespaces] [answered]
When connected to codespaces (browser or VS Code instance), sometimes I run a command in the VS Code terminal window, and when I return the codespace is stopped due to inactivity. When I resume the codespace I need to see the state and output of the terminal, and sometimes need to run additional commands in the terminal that depend on the environment as-is. 

What codespaces does when I resume from sleep/stopped state is that the terminal window is a completely new instance and I have lost the history of what was run and I cannot see the output of the last command.

It would be nice to be able to resume the codespace from sleep/stopped state and have the terminal also restored to its original state.

Otherwise this is an awesome tool. ❤️

**Comment by tanmayeekamath** [comment, marked as answer]
Hey @brooke-hamilton, thanks for sharing the feedback. I have captured this and added as a plus 1 to our existing thinking around retaining terminal history on reconnect.

**Reply by iloveitaly** [reply]
A 👍 from my end: this would be a helpful feature.

**Reply by alancarwile** [reply]
I'd like to +1 this, but feel like I should do so on the topic to which @tanmayeekamath refers.
- Please point us to that one so we can +1 the right thing.
- Any update on status?
I showed my students that commands and output were saved. But it was only because I reopened the codespace very quickly.  After an hour or two the content of the session is gone.

**Comment by brahn** [comment]
Looking for this as well!  Any update @tanmayeekamath?

PS Hi @iloveitaly :-)

**Comment by jakeshi** [comment]
Any update on this?

**Comment by antonibertel** [comment]
It's a must-have. 

I ended up installing the Terminal Keeper, creating one profile that kills all previous sessions (if any are running in the background), and another profile that is running my apps

`"commands": ["for dir in */; do fuser -k \"$dir\"; done"],`
